.github/workflows/publish.yaml

```diff
@@ -1,24 +1,19 @@
 name: Publish to PyPi and create GitHub Release
 
 on:
```

Comment:
I think we need to make sure the tests are running only if Test pasted otherwise we can have a release even if tests failed.

Reply:
good point